File: 153.find.minimum.in.rotated.sorted.array.rb
# @param {Integer[]} nums
# @return {Integer}
def find_min(nums)
    lo, hi = 0, nums.length-1
    while lo < hi
        mid = (lo + hi) >> 1
        if nums[mid] >= nums[hi]
            lo = mid + 1
        else
            hi = mid
        end
    end
    return nums[lo]
end

File: 153. Find Minimum in Rotated Sorted Array.go
func findMin(nums []int) int {
	n := len(nums)
	l, r := 0, n-1
	for l < r {
		m := (l + r) / 2
		if nums[m] >= nums[l] && nums[m] >= nums[r] {
			l = m + 1
		} else {
			r = m
		}
	}
	return nums[l]
}